src/main/java/com/firebolt/jdbc/connection/FireboltConnection.java

```diff
@@ -438,11 +437,15 @@ public boolean isValid(int timeout) throws SQLException {
 
 	private void validateConnection(FireboltProperties fireboltProperties, boolean ignoreToManyRequestsError, boolean isInternalRequest)
 			throws SQLException {
-		FireboltProperties propertiesCopy = FireboltProperties.copy(fireboltProperties);
+		HashMap<String, String> runtimeProperties = new HashMap<>(fireboltProperties.getRuntimeAdditionalProperties());
 		if (isInternalRequest) {
-			propertiesCopy.addProperty("auto_start_stop_control", "ignore");
+			runtimeProperties.put("auto_start_stop_control", "ignore");
+		}
+		var propertiesBuilder = fireboltProperties.toBuilder().runtimeAdditionalProperties(runtimeProperties);
+		if (getSessionProperties().isValidateOnSystemEngine()) {
+			propertiesBuilder.compress(false).engine(null).systemEngine(true);
```

Comment:
Why do we have `compress(false)` defined here? Is there a special behaviour on system engine?

Reply:
For system engine we automatically disable compress in JDBC.
https://github.com/firebolt-db/jdbc/blob/master/src/main/java/com/firebolt/jdbc/connection/settings/FireboltProperties.java#L97
I'm not sure of the reasons, all that I can say is that system engine doesn't work with compres as tests have shown. We receive an expected end of stream, and I failed ot understand the reason behind it